### PR TITLE
Add pytorch/xla r2.3-rc2 tests

### PR DESCRIPTION
**Pull request by lsy323**
# Description

Add pytorch/xla r2.3-rc2 tests

- The jsonnet test files are copied from nightly test.
- Set running schedule to None, so the new test sets won't run periodically. It can be triggered manually.
- Updated the script to install 2.3-rc2 torch, torch/xla whl and the compatible torchvision.

# Tests

Testing with setting up local Airflow. But failed with unknown error during `setup` stage.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by zpcore** [comment]
Thanks for bringing up the 2.3 tests. LGTM!

Can you run `bash scripts/upload-tests.sh gs://us-central1-ml-automation-s-bc6c8818-bucket/dags` and check on [airflow dev](https://08398f98ca104983a06a424b851a390d-dot-us-central1.composer.googleusercontent.com/home) to see if any issue pop up?

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (zpcore) [inline dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:27]: Are we updating from 3.8 to 3.10 build of pytorch?

Comment (zpcore) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:27]: Hmm, just noticed that in nightly, the PT is specified using 3.8 while the torch_xla is 3.10 build. You are good here!

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (zpcore) [inline dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:104]: I think `torch_xla[tpuvm]` should be sufficient. @will-cromar to confirm.

Comment (will-cromar) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:104]: No, we still need the GCS path since the  release is not uploaded to PyPI yet.

Speaking of PyPI, can you swap in the `tpu` extra dependency instead? We have to remove `tpuvm` from the build or PyPI will reject it.

Something like this should work: `'torch_xla[tpu] @ https://storage.googleapis.com/libtpu-releases/index.html @ https://storage.googleapis.com/pytorch-xla-releases/wheels/tpuvm/torch_xla-2.3.0rc2-cp310-cp310-linux_x86_64.whl' -f https://storage.googleapis.com/libtpu-releases/index.html`

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (will-cromar) [inline dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:101]: For torchvision and torch, can you install the upstream RCs instead? There will be instructions in the release announcement.

Comment (lsy323) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:101]: There is no upstream RC whl released yet. I tried `pip install torch==2.3.0`

Comment (will-cromar) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:101]: PyTorch releases RCs through their private index. There are examples commands here: https://dev-discuss.pytorch.org/t/pytorch-2-3-release-branch-cut-for-pytorch-core-is-completed/1939

It looks like PyTorch is there, but not `torchvision` (yet)

File: dags/pytorch_xla/r2_3.py
Comment (zpcore) [inline dags/pytorch_xla/r2_3.py:24]: Can you update this to once per day?

Comment (lsy323) [reply dags/pytorch_xla/r2_3.py:24]: As @will-cromar suggested, we set `None` here and we can trigger manually. tests for rc doesn't need to be run daily, since they tag is immutable

Comment (zpcore) [reply dags/pytorch_xla/r2_3.py:24]: Oh, I see, make sense to me!

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (lsy323) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:104]: Sounds good, updating to `pip install torch_xla[tpu] -f https://storage.googleapis.com/libtpu-releases/index.html`

File: dags/pytorch_xla/r2_3.py
Comment (will-cromar) [reply dags/pytorch_xla/r2_3.py:24]: Shouldn't we only run this manually when the wheel updates?

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (lsy323) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet]: Updated

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (lsy323) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:101]: Sounds good, updated to `pip install torch==2.3.0 https://download.pytorch.org/whl/test/cpu`

**Comment by lsy323** [comment]
Thanks @will-cromar @zpcore for reviewing. I'll test locally again before landing.

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (will-cromar) [inline dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:143]: Sorry that this is duplicated lol

I really need to clean up these configs now...

**Review by will-cromar — APPROVED** [review]
LGTM if test run passes

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (lsy323) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:143]: Thanks @will-cromar, removing these [lines](https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/198/files/5b327be8874d7bc34bf23e7acfb09c94e67af6c9#diff-be01cb501c71f2b4a7c669041385313192a6802fdda100de72e2e64f1b765895R142-R148) ok?

**Comment by lsy323** [comment]
Hitting an error when running GPU tests, @zpcore pointed out it's naming issue. We need to use `2-3` instead of `2_3`

File: dags/legacy_test/tests/pytorch/r2.3/common.libsonnet
Comment (will-cromar) [reply dags/legacy_test/tests/pytorch/r2.3/common.libsonnet:143]: You can skip installing `torch` and `torch_xla` since this code runs in our docker image that will have both. We will need to install `torchvision` and clone `xla`

**Pull request merged**
